src/graphics/Screen.cpp

```diff
@@ -2297,6 +2298,12 @@ void Screen::handleOnPress()
     if (scanAndSelectInput != nullptr && scanAndSelectInput->dismissCannedMessageFrame())
         return;
 
+    // Don't transition away from canned messages if it is active
+    if (cannedMessageModule->shouldDraw()) {
+        lastScreenTransition = millis();
+        return;
+    }
+
```

Comment:
Would this prevent users from being able to escape the canned message frame with the user-button? I'm not sure, but I think that might be an important behavior for a rotary-encoder + user button build.

Reply:
The canned message module will close if a InputEventChar_CANCEL event is received, as normal. Or are you referring to where the user scrolls with the encoder, then doesn't select a message? In that case the CannedMessageModule INACTIVATE_AFTER_MS timeout will return to the carousel after 20 secs. This PR fixes two bugs that prevented this timeout from working correctly.

To be clear, the current method of using the Screen transition as a canned message timeout has the following issues:

- Screen transition timer does not reset on user input, leading to erratic timeouts.
- CannedMessageModule is still in CANNED_MESSAGE_RUN_STATE_ACTIVE state after timeout which breaks normal carousel functions: Left/Right no longer moves between frames, and Select sends the previously chosen but hidden canned message instead of moving to the next frame.

Reply:
Full disclosure: I'm not *super* familiar with the canned message module.

> Or are you referring to where the user scrolls with the encoder, then doesn't select a message?

That's the case I was wondering about. Just when I was playing with a test setup a while back, I remember being able to exit the canned message frame at will by pressing the user button. If I understand you right, the module frame closes itself after several seconds of no input?

Hopefully tonight I'll have a chance to build it and try out a few different input methods. I might not be the best reviewer here sorry. I don't have a standalone keyboard myself, so I can't test it as fix, but I can at least give it a run and see if anything odd pops up.